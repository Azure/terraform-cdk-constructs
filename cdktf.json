{
  "language": "typescript",
  "app": "npx ts-node ./src/examples/AzureResourceGroup.ts",
  "terraformProviders": ["Azure/azapi@~> 0.6.0"],
  "projectId": "1b5f16bd-67ac-4af6-a33f-7bcafff141eb",
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}