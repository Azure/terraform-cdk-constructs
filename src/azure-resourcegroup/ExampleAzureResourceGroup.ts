import { AzureResourceGroup } from '.';
import { App, TerraformStack} from "cdktf";
import {AzurermProvider} from "@cdktf/provider-azurerm/lib/provider";
import { Construct } from 'constructs';
import { generateRandomString } from '../util/randomString';

const rndName = generateRandomString(10);

const app = new App();

export class exampleAzureResourceGroup extends TerraformStack {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    new AzurermProvider(this, "azure", {
      features: {},
    });
  
    new AzureResourceGroup(this, 'testRG', {
      name: `rg-${rndName}`,
      location: 'eastus',
      tags: {
          name: 'test',
          Env: "NonProd",
      },
      ignoreChanges: ['tags["Environment"]'],
      
    });

  }
}


new exampleAzureResourceGroup(app, "testAzureResourceGroup");
app.synth();