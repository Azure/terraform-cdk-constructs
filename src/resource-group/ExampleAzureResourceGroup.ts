import { AzureResourceGroup } from '.';
import { App, TerraformStack} from "cdktf";
import {AzurermProvider} from "@cdktf/provider-azurerm/lib/provider";
import { Construct } from 'constructs';


const app = new App();

export class exampleAzureResourceGroup extends TerraformStack {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    new AzurermProvider(this, "azureFeature", {
      features: {},
    });
  
    new AzureResourceGroup(this, 'testRG', {
      name: 'rg-test',
      location: 'eastus',
      tags: {
          name: 'test',
          Env: "NonProd",
      },
      
    });

  }
}


new exampleAzureResourceGroup(app, "testAzureResourceGroup");
app.synth();