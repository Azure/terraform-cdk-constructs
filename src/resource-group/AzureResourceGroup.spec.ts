import { Testing, TerraformStack} from 'cdktf';
import { exampleAzureResourceGroup} from './ExampleAzureResourceGroup'
import 'cdktf/lib/testing/adapters/jest';


describe('AzureResourceGroup-Snapshot', () => {
  it('renders a AzureResourceGroup and checks snapshot', () => {

    const synthed = Testing.synthScope((stack) => {
      new exampleAzureResourceGroup(stack, "testAzureResourceGroup");
    });
    expect(synthed).toMatchSnapshot();
  });
});


describe("AzureResourceGroup-Terraform", () => {
  it("check if the produced terraform configuration is valid", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");
    
    Testing.synthScope((stack) => {
      new exampleAzureResourceGroup(stack, "testAzureResourceGroup");
    });

    // We need to do a full synth to validate the terraform configuration
    expect(Testing.fullSynth(stack)).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");

    Testing.synthScope((stack) => {
      new exampleAzureResourceGroup(stack, "testAzureResourceGroup");
    });


    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(stack)).toPlanSuccessfully();
  });
});
