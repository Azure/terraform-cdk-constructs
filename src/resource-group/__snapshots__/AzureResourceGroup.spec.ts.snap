// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzureResourceGroup-Snapshot renders a AzureResourceGroup and checks snapshot 1`] = `
"{
}"
`;
